extension/jaegerremotesampling/internal/remote_strategy_store.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal"
+
+import (
+	"context"
+	"fmt"
+
+	grpcstore "github.com/jaegertracing/jaeger/cmd/agent/app/configmanager/grpc"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type grpcRemoteStrategyStore struct {
+	conn            *grpc.ClientConn
+	delegate        *grpcstore.SamplingManager
+	headerAdditions map[string]configopaque.String
```

Comment:
Does this have to be thread safe in any form?

Reply:
because this is only ever _written_ to at init time via the config, I think it's safe because golang map reads are threadsafe (even if, for instance, several threads are invoking the "context-enhancing" code path concurrently)